lib/v2/ptu/news/news.js

```diff
@@ -0,0 +1,111 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const timezone = require('@/utils/timezone');
+const { parseDate } = require('@/utils/parse-date');
+
+const baseUrl = 'https://news.ptu.edu.cn/';
+
+
+async function getPageSize(ctx, url) {
+    const response = await ctx.cache.tryGet(url, async () => {
+        const { data } = await got({ method: 'get', url }, { https: { rejectUnauthorized: false } });
+        return data;
+    });
+    const $ = cheerio.load(response);
+    return parseInt($(".p_next").find("a").attr("href").match("\\d+")[0]) + 1;
+}
+
+function getPageUrl(url, pageId, pageSize) {
+    pageId = parseInt(pageId);
+    if (pageId && pageId !== 1) {
+        const currentPagePath = `/${pageSize - pageId + 1}.htm`;
+        const pageUrl = url.slice(0, -4) + currentPagePath;
+        return pageUrl;
+    }
+    return url;
+}
+
+const getArticleContent = async (ctx, link) => {
+    const response = await ctx.cache.tryGet(link, async () => {
+        const { data } = await got({ method: 'get', url: link }, { https: { rejectUnauthorized: false } });
+        return data;
+    });
+    const $ = cheerio.load(response);
+    const publisher = $('.bar').text().match("发布者：(.*)发布时间")[1].trim();
+    const responsibleEditor = $('.bar').text().match("责任编辑：(.*)发布者")[1].trim();
+
+    // Get the last match result
+    let authorInArticle = [...$('.v_news_content').text().matchAll(/[(（]([\W]{3,20}?[\W\s]{2,30}?)[）)]/g)].pop();
+    // console.log (authorInArticle)
+    authorInArticle = authorInArticle !== undefined ? authorInArticle[authorInArticle.length - 1].replace("供稿", "").replace("：", "") : "";
+    const author = authorInArticle !== "" ? authorInArticle
+                  : publisher !== "" ? publisher
+                  : responsibleEditor;
+    return {
+         author,
+         article: $('.v_news_content').html() };
+};
```

Comment:
Please cache the returned data instead of HTML response

Reply:
Done.